pkg/deploy/api/v1beta1/types.go

```diff
@@ -180,6 +180,12 @@ const (
 	// DeploymentConfigLabel is the name of a label used to correlate a deployment with the
 	// DeploymentConfigs on which the deployment is based.
 	DeploymentConfigLabel = "deploymentconfig"
+	// DeploymentStatusReasonAnnotation represents the reason for deployment being in a given state
+	// Used for specifying the reason for cancellation or failure of a deployment
+	DeploymentStatusReasonAnnotation = "deploymentStatusReason"
```

Comment:
Well, this is an open question for me. There is nothing preventing us from namespacing any new annotations that we add, even for v1beta1. Decided to keep them as-is for consistency. 

But now that I think about it, if we add more annotations over time, we'll have to keep maintaining the entries in the annotation map. So, maybe I'll namespace the new annotations even for v1beta1 if you feel that's ok.